File: C language/programs/sort/linear_search.c
#include<stdio.h>
int main()
{
    int n,num,i;
    printf("Enter the number of Elements: ");
    scanf("%d",&n);
    int arr[n];
    for(i=0;i<n;i++)
    {
        printf("Enter the element number %d : ",i+1);
        scanf("%d",&arr[i]);
    }
    printf("Enter the element to search: ");
    scanf("%d",&num);
    for(i=0;i<n;i++){
        if(num==arr[i])
            printf("Element \"%d\" found at index %d.\n",num,i);
    }

    return 0;
}

File: Java/Algorithms/Array/searching/LinearSearch.java
import java.util.Scanner;

class LinearSearch{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = sc.nextInt();
        int[] array = new int[n];
        getArray(array, n); 
        int index = lsearch(array, n);   
        if(index == -1){
            System.out.println("Key not found.");
        }
        else{
            System.out.println("Key found at index " + index + ".");
        }
    }

    public static int lsearch(int[] array, int n){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the key: ");
        int key = sc.nextInt();

        for(int i = 0; i < n; i++){
            if(array[i] == key){
                return(i);
            }
        }

        return(-1);
    }

    public static void getArray(int[] array, int n){
        Scanner sc = new Scanner(System.in);
        for(int i = 0; i < n; i++){
            System.out.print("Enter element " + (i + 1) + " : ");
            array[i] = sc.nextInt();
        }
    }

    public static void printArray(int[] array, int n){
        System.out.println();
        for(int i = 0; i < n; i++){
            System.out.print(array[i] + " ");
        }
    }
}